### ISSUE TITLE
ResourceGraph missing parameter

### ISSUE THREAD

**Issue by xuzhang3** [Azure/azure-sdk-for-go#8559]
### Bug Report

- SDK version v41.2.0
 

- What happened?
  - struct ```GraphQueryResource``` defined in ```services\preview\resourcegraph\mgmt\2018-09-01\resourcegraph\models.go```  missing required parameter ```location```
- What did you expect or want to happen?
  - add  required parameter ```location``` to  ```GraphQueryResource```
- How can we reproduce it?
   -  Use ResourceGraph CreateOrUpdatre API: ```https://docs.microsoft.com/en-us/rest/api/azureresourcegraph/resourcegraph(2018-09-01-preview)/graphquery/createorupdate#code-try-0```. 
   - Request body 
    ```json
    {
       
       "properties" : {
              "query": "Resources | where type=~ 'microsoft.compute/virtualmachinescalesets' |  order by        Capacity desc"
       }
     }
    ```
   - Response Body:
    ```json
      {
          "error": {
           "code": "LocationRequired",
                "message": "The location property is required for this definition."
           }
       }
     ```

- Anything we should know about your environment.

**Comment by ArcturusZhang** [comment]
Relative swagger: https://github.com/Azure/azure-rest-api-specs/blob/d1ae5dd8ae6803e1ca57dfba5934389df76fd77f/specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/preview/2018-09-01-preview/graphquery.json#L246
There should be a `location` parameter in the `Resource` definition and it should be required.

**Comment by demyanenko** [comment]
Thank you for reporting the issue. PR is out with the fix: https://github.com/Azure/azure-rest-api-specs/pull/9216/

**Comment by johnthcall** [comment]
This has also been brought up in Azure/azure-rest-api-specs#9136 and should be fixed with Azure/azure-rest-api-specs#9216

**Comment by ArcturusZhang** [comment]
Thanks for the help @demyanenko and @johnthcall 
Close this issue in favor of https://github.com/Azure/azure-rest-api-specs/issues/9136
